contrib/log_correlation/stackdriver/src/main/java/io/opencensus/contrib/logcorrelation/stackdriver/OpenCensusTraceLoggingEnhancer.java

```diff
@@ -24,35 +24,114 @@
 import io.opencensus.trace.TraceId;
 import io.opencensus.trace.Tracer;
 import io.opencensus.trace.Tracing;
+import java.util.logging.LogManager;
 
 /**
  * Stackdriver {@link LoggingEnhancer} that adds OpenCensus tracing data to log entries.
  *
  * <p>This feature is currently experimental.
  */
-
-// TODO(sebright): Allow configuring the LoggingEnhancer with Java properties. One property should
-// control the types of spans to correlate, with these three choices:
-// - off
-// - only add tracing data for sampled spans
-// - add tracing data for all spans
 @ExperimentalApi
 public final class OpenCensusTraceLoggingEnhancer implements LoggingEnhancer {
-  private static final String SAMPLED_KEY = "sampled";
+  private static final String SAMPLED_LABEL_KEY = "sampled";
+  private static final SpanSelection DEFAULT_SPAN_SELECTION = SpanSelection.SAMPLED_SPANS;
+
+  /** Name of the property that defines the {@link SpanSelection}. The value is {@value}. */
+  public static final String SPAN_SELECTION_PROPERTY_NAME =
+      "io.opencensus.contrib.logcorrelation.stackdriver."
+          + "OpenCensusTraceLoggingEnhancer.spanSelection";
 
   private static final Tracer tracer = Tracing.getTracer();
 
-  /** Constructor to be called by Stackdriver logging. */
-  public OpenCensusTraceLoggingEnhancer() {}
+  private final SpanSelection spanSelection;
+
+  /** How to decide whether to add tracing data from the current span to a log entry. */
+  public enum SpanSelection {
+
+    /**
+     * Never add tracing data to log entries. This constant disables the log correlation feature.
+     */
+    NO_SPANS,
+
+    /** Add tracing data to a log entry iff the current span is sampled. This is the default. */
+    SAMPLED_SPANS,
```

Comment:
what's the use case for this btw? I've personally not seen this arise (only sampled spans), and folks like @nicmunroe @bplotnick tend to suggest always trace ids. attaching nic's latest neat blog on it.

https://medium.com/nikeengineering/hit-the-ground-running-with-distributed-tracing-core-concepts-ff5ad47c7058

Reply:
Thanks for the link!  That is really helpful for understanding the use case of exporting tracing data primarily through logs.  I added the SAMPLED_SPANS option for the case where users are already using Stackdriver Trace and want to use log messages similarly to annotations, where the tracing data for non-sampled spans would be less useful.  It sounds like ALL_SPANS would be a better default, though.  I updated the PR.